contractsSol5/KyberFeeHandler.sol

```diff
@@ -303,46 +310,53 @@ contract KyberFeeHandler is IKyberFeeHandler, Utils4 {
         emit RewardsRemovedToBurn(epoch, rewardAmount);
     }
 
-    function encodeBRRData(uint _reward, uint _rebate, uint _epoch, uint _expiryBlock) public pure returns (uint) {
-        return (((((_reward << BITS_PER_PARAM) + _rebate) << BITS_PER_PARAM) + _epoch) << BITS_PER_PARAM) + _expiryBlock;
-    }
-
-    function decodeBRRData() public view returns(uint rewardBPS, uint rebateBPS, uint expiryBlock, uint epoch) {
-        expiryBlock = brrAndEpochData & (1 << BITS_PER_PARAM) - 1;
-        epoch = (brrAndEpochData / (1 << BITS_PER_PARAM)) & (1 << BITS_PER_PARAM) - 1;
-        rebateBPS = (brrAndEpochData / (1 << (2 * BITS_PER_PARAM))) & (1 << BITS_PER_PARAM) - 1;
-        rewardBPS = (brrAndEpochData / (1 << (3 * BITS_PER_PARAM))) & (1 << BITS_PER_PARAM) - 1;
-        return (rewardBPS, rebateBPS, expiryBlock, epoch);
+    function readBRRData() public view returns(uint rewardBps, uint rebateBps, uint expiryBlock, uint epoch) {
+        rewardBps = uint(brrAndEpochData.rewardBps);
+        rebateBps = uint(brrAndEpochData.rebateBps);
+        epoch = uint(brrAndEpochData.epoch);
+        expiryBlock = uint(brrAndEpochData.expiryBlock);
     }
 
-    event BRRUpdated(uint rewardBPS, uint rebateBPS, uint burnBPS, uint expiryBlock, uint epoch);
+    event BRRUpdated(uint rewardBps, uint rebateBps, uint burnBps, uint expiryBlock, uint epoch);
 
-    function getBRR() public returns(uint rewardBPS, uint rebateBPS, uint epoch) {
+    function getBRR() public returns(uint rewardBps, uint rebateBps, uint epoch) {
         uint expiryBlock;
-        (rewardBPS, rebateBPS, expiryBlock, epoch) = decodeBRRData();
+        (rewardBps, rebateBps, expiryBlock, epoch) = readBRRData();
 
           // Check current block number
         if (block.number > expiryBlock && kyberDAO != IKyberDAO(0)) {
-            uint burnBPS;
+            uint burnBps;
 
-            (burnBPS, rewardBPS, rebateBPS, epoch, expiryBlock) = kyberDAO.getLatestBRRData();
+            (burnBps, rewardBps, rebateBps, epoch, expiryBlock) = kyberDAO.getLatestBRRData();
+            require(burnBps + rewardBps + rebateBps == BPS, "Bad BRR values");
```

Comment:
could bad DAO return BRR that causes overflow here? May be can add a test with bad DAO for the overflow.

Reply:
Yes, it's possible to have overflow.
setMockBRR(10000, 1)
burnBPS = BPS - 10000 - 1 = 2**256 - 1
The check will still pass, but the values are invalid.

Will need more checks here.